### ISSUE TITLE
Local roomlist sorting causes a spasm of sorting whenever you open the app.

### ISSUE THREAD

**Issue by ara4n** [element-hq/element-x-ios#3009]
### Steps to reproduce

1. Open the app for the first time in a bit (perhaps after the SS session has expired?)
2. Watch as it syncs
3. See that after the syncing spinner goes away, it incrementally refreshes the roomlist very rapidly as it reorders it, over the course of about 100ms, causing a manic spasm of reordering which is ugly, feels flakey and unpredictable, and is a regression from the previous behaviour.

#### What did you expect?

Once synced, the visible roomlist should reorder atomically to the new ordering, giving a sense of stability and control, rather than doing a frenetic animation of intermediary state.

### Application version

642

### Will you send logs?

Yes

**Comment by ara4n** [comment]
this has improved in build 644, but still happens to some extent

**Comment by manuroe** [comment]
@ara4n do you still observe this bug with your account?

**Comment by stefanceriu** [comment]
I don't believe this is a problem anymore, we flushed out the local room sorting bugs since then.
